samples/client/petstore/go-experimental/go-petstore/model_dog.go

```diff
@@ -16,59 +16,10 @@ import (
 
 // Dog struct for Dog
 type Dog struct {
-	ClassName string `json:"className"`
-	Color *string `json:"color,omitempty"`
+	Animal
```

Comment:
This is very cool. I'm not a pro in Go by any means, but I've just learned about field promotion in anonymous nested structs. ;)